controllers/topolvm_controller.go

```diff
@@ -0,0 +1,258 @@
+package controllers
+
+import (
+	"context"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	controllerName = "topolvm-controller"
+)
+
+type controllerPlugin struct{}
+
+// controllerPlugin unit satisfies resourceManager interface
+var _ resourceManager = controllerPlugin{}
+
+func (c controllerPlugin) getName() string {
+	return controllerName
+}
+
+func (c controllerPlugin) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	// get the desired state of topolvm controller deployment
+	controllerDeployment := getControllerDeployment(lvmCluster)
+	result, err := cutil.CreateOrUpdate(ctx, r.Client, controllerDeployment, func() error {
```

Comment:
Better to update only if there is a change in the spec.

Reply:
- Topolvm Controller deployment is resistant to LVM Cluster CR Updates
- After first creation the result will mostly be `Unchanged`
- Current implementation is used only to facilitate upgrade due to change in images, if both desired and existing state is same then it'll be `Unchanged`